core/src/test/scala/integration/kafka/server/ReplicaManagerTest.scala

```diff
@@ -0,0 +1,61 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package integration.kafka.server
+
+import kafka.server.KafkaConfig
+import kafka.utils.TestUtils
+import kafka.zk.ZooKeeperTestHarness
+import org.apache.kafka.common.TopicPartition
+import org.apache.kafka.common.utils.MockTime
+import org.junit.Test
+import org.junit.Assert._
+
+class ReplicaManagerTest extends ZooKeeperTestHarness {
+  @Test
+  def testMaybePropagateIsrChanges(): Unit = {
+    val config = KafkaConfig.fromProps(TestUtils.createBrokerConfig(1, zkConnect))
+    val mockTime = new MockTime(10000)
+    val server = TestUtils.createServer(config, mockTime)
+
+    // Topic name which is the maximum length of 249.
+    val largeTopicName = ("topic-" padTo(249, "x")).mkString
+
+    // 5000 partitions. Large partition numbers chosen to make the serialized values as long as possible.
+    (10000 to 15000)
+      .map(n => new TopicPartition(largeTopicName, n))
+      .foreach(server.replicaManager.recordIsrChange)
+
+    server.replicaManager.maybePropagateIsrChanges()
+
+    val isrChangeNotificationQueuePath = "/isr_change_notification"
+    val node0 = "isr_change_0000000000"
+    val node1 = "isr_change_0000000001"
+
+    assertEquals(
+      zkClient.getChildren(isrChangeNotificationQueuePath).toSet,
+      Set(node0, node1))
+
+    val (_, stat0) = zkClient.getDataAndStat(isrChangeNotificationQueuePath + "/" + node0)
+    val (_, stat1) = zkClient.getDataAndStat(isrChangeNotificationQueuePath + "/" + node1)
+
+    assertEquals(840028, stat0.getDataLength)
+    assertEquals(560308, stat1.getDataLength)
```

Comment:
This will make the test a bit brittle. Do we want to check for the exact value or is a threshold better? Either way, we should probably add a comment explaining what these values mean and what to do when they invariably start failing.

Reply:
Good point I moved the test to the existing `ReplicaManagerTest`.